### ISSUE TITLE
Multi-words application name

### ISSUE THREAD

**Issue by AndrewGri** [qlikperf/TSEEQ#7]
I need to have ability to construct application name with several words. For example, "**Sales Department Activity**". In current version, I have to redefine in each application from ETL sequence the vApp variable and the vPurpose variable values. This comes from the default ' ' character as separator. I'd like to have these variables to be defined automatically, so I modified the _Set_vApp_dynamically.txt_ and _Set_vPurpose_dynamically.txt_. The results are:

**Set_vApp_dynamically.txt**
```
IF len(trim(DocumentTitle() ) ) >  0 THEN 		
	LET vApp = 	subfield(DocumentTitle(), '@', 1);  // The Title will always be set in Qlik Sense, but only sometimes in QlikView.
ELSE 
	LET vApp = subfield(DocumentName(), '@', 1) ;	// If the Title is not set, use the file name, which is returned by DocumentName().
ENDIF

IF upper(vQlikProduct) = 'VIEW' THEN
	LET vApp = upper(vApp);
ENDIF

/****************************************************************************************************************************************
	File: 		Set_vApp_dynamically.txt
	Included by:	All QVFs and QVWs. 
		
	Author:		Jeff R. Robbins
	Email:		jrs@qlik.com; jr@QlikPerf.com
	
	Version:	3.1.7
	Date:		2018-12-09
	
	This program is provided in the hope that it will be useful,
	but WITHOUT ANY WARRANTY; without even the implied warranty of
	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  
	
	Purpose:		Set the vApp variable dynamically, based on the output of DocumentTitle() or DocumentName().
****************************************************************************************************************************************/
```

**Set_vPurpose_dynamically.txt**

```

IF len(trim(DocumentTitle() ) ) >  0 THEN 		
  	// The Title will always be set in Qlik Sense, but only sometimes in QlikView.
  	LET vPurpose = 	subfield ( replace( replace(upper(subfield(DocumentTitle(), '@', 2)), 'DATAMODEL', 'LOAD'), 'DASHBOARD', 'APP'), '(', 1);
ELSE 
	// If the Title is not set, use the file name, which is returned by DocumentName()
	LET vPurpose = replace( replace( upper(subfield(subfield(DocumentName(), '@', 2), '.', 1)), 'DATAMODEL', 'LOAD'), 'DASHBOARD', 'APP') ;	
ENDIF
/****************************************************************************************************************************************
	File: 		Set_vPurpose_dynamically.txt
	Included by:	All QVFs and QVWs. 
		
	Author:		Jeff R. Robbins
	Email:		jrs@qlik.com; jr@QlikPerf.com
	
	Version:	3.1.7b
	Date:		2018-02-19
	
	This program is provided in the hope that it will be useful,
	but WITHOUT ANY WARRANTY; without even the implied warranty of
	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  
	
	Purpose:		Set the vPurpose variable dynamically, based on the output of DocumentTitle() or DocumentName().
****************************************************************************************************************************************/

```

I also renamed ETL-application files from the _TSEEQ Sales Sample_ folder to Sales@Extract, Sales@Transform, Sales@Dashboard.

And finally I modified the folder creation bat-file:

```
...

:CopyQVFs
	
	Copy  "%vTSEEQSalesFolder%1_Extract\Sales@Extract.QVF"  	"%vNewAppFolder%\1_Extract\%vNewAppName%@Extract.QVF"
	Copy  "%vTSEEQSalesFolder%2_Transform\Sales@Transform.QVF" 	"%vNewAppFolder%\2_Transform\%vNewAppName%@Transform.QVF"
 	REM Copy  "%vTSEEQSalesFolder%3_Load\Sales DataModel.QVF"  		"%vNewAppFolder%\3_Load\%vNewAppName% DataModel.QVF"
 	Copy  "%vTSEEQSalesFolder%4_App\Sales@Dashboard.QVF"  		"%vNewAppFolder%\4_App\%vNewAppName%@Dashboard.QVF"
 	Copy  "%vTSEEQSalesFolder%4_App\Sales Datamodel and Dashboard Combined.QVF"  "%vNewAppFolder%\4_App\%vNewAppName% Sales Datamodel and Dashboard Combined.QVF"
...
```

So, I'm able to use the multi-words application names "**Long Application Name**" and these names are correctly processed during vApp and vPurpose automatically set values.

P.S. I did not modify comments, but I'm sure it should be done :)

**Comment by qlikperf** [comment]
@AndrewGri , you are starting to become annoying!!!   :-)  You are responsible for a very large percentage of _**TSEEQ**_ issues.  How large of a percentage?  Sorry, I'm **not** good at mathematics.  That's why I joined QlikTech back in 2007.  QlikTech (ok, now it's just called Qlik) provides me free unlimited access to software that does math for me. 

Please do feel free to email me; you know the address...  I would love to hear from you directly.  Quite honestly, _**TSEEQ**_ adoption is not very high...   I would like to understand what about _**TSEEQ**_ you like and find useful.  And you probably have co-workers that do _**not**_ use _**TSEEQ**_.  Why not?  I  would love to hear the reasons!

**Comment by qlikperf** [comment]
Hi @AndrewGri ,  thanks for this great issue report.

Would per chance be willing and able to become a **_TSEEQ_** contributor and open a pull request for your proposed changes?   Thus far, I've been the only **_TSEEQ_** contributor, and it has been a very lonely experience.

If you could please become a contributor, we would get some much needed momentum in _**TSEEQ**_ adoption.  

Thanks,

J. R.

**Comment by AndrewGri** [comment]
Hi Jeff, thanks for your answer to my issue report.

I'm a data engineer in a company, and now we're on the way to implementing
BI solution. Now we're on the stage when different applications load data
from database, and it's difficult to understand, which app uses which qvd,
what transformations take place and so on. That's why I decided to have a
look and lately to use the *TSEEQ*.

I saw the description of using TSEEQ in this artlice
<https://qliksense.ivan-shamaev.ru/qlik-sense-etl-data-load-script-basic/#TSEEQ_The_Structured_ETL_Engine_for_Qlik_Sense>
(yes,
I'm from Russia:)  ).

I like very much the way TSEEQ give me ability to configure extract and
transform steps and monitor these steps execution.

I'm absolutely new in a GitHub project contribution, but I'm ready to help
you in this project.

Best regards,
Andrew

чт, 10 дек. 2020 г. в 01:25, Jeff R. Robbins <notifications@github.com>:

**Comment by qlikperf** [comment]
Hi Andrew,

I have just published release **v4.1.3** that incorporates your ideas and contributions, with some slight changes.  Specifically, please note that I am keeping the **_TSEEQ_** Sales sample files as in prior releases, with spaces, rather than the @ symbol, as the word separator.   

However, you can now specify an _optional_ **MULTIWORDS** parameter to the following file:  **Create New Qlik Sense App Folder from SEEQ Sales Sample.bat**.

The **MULTIWORDS** parameter will name the new QVFs in the new App folder as per your request, using the @ symbol as the word separator in the file names. 

Invoke as follows:
    c:\TSEEQ\Utility\Folder Creation BAT>"Create New Qlik Sense App Folder from SEEQ Sales Sample.bat" MULTIWORDS

Please note that in the command line, MULTIWORDS should _not_ be surrounded by quotes. 

If the MULTIWORDS parameter is not included, then the new QVFs in the new App folder will be named as in previous releases, with the space character as the word separator in the file names. 

The release v4.1.3 also includes changes to the following 2 files, in order handle the new case where the @ symbol is used as the word separator:

1. Set_vApp_dynamically.txt
2. Set_vPurpose_dynamically.txt
